File: 230-kth-smallest-element-in-a-bst/230-kth-smallest-element-in-a-bst.cpp
class Solution {
public:
    int kthSmallest(TreeNode* root, int k) {
        priority_queue<int> maxHeap;
        queue<TreeNode*> q;
        q.push(root);
        while(!q.empty())
        {
            TreeNode* head = q.front();
            q.pop();
            if(head->left)
                q.push(head->left);
            if(head->right)
                q.push(head->right);
            maxHeap.push(head->val);
            if(maxHeap.size()>k)
                maxHeap.pop();
        }
        return maxHeap.top();
    }
};

File: 230-kth-smallest-element-in-a-bst/230-kth-smallest-element-in-a-bst.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
   
    int count = 0;
    
    public int kthSmallest(TreeNode root, int k) {
        ArrayList<Integer> ans = new ArrayList<>();
        inorder(root, k, ans, count);
        return ans.get(k-1);
    }
    
    public void inorder(TreeNode root, int k, ArrayList<Integer> ans , int count){
        if(root == null) {
            return;
        }
        inorder(root.left, k,ans, count);
        ans.add(root.val);
        inorder(root.right,k,ans, count);
        
    }
}